Model/Api/ShippingMethods.php

```diff
@@ -430,7 +458,7 @@ public function shippingEstimation($quote, $addressData)
 
             $cacheIdentifier = $parentQuoteId.'_'.round($quote->getSubtotal()*100).'_'.
                 $addressData['country_code']. '_'.$addressData['region'].'_'.$addressData['postal_code']. '_'.
-                @$addressData['street_address1'].'_'.@$addressData['street_address2'];
+                @$addressData['street_address1'].'_'.@$addressData['street_address2'].'_'.$externalData;
```

Comment:
The `@` - will produce a slow performance and also dismiss error. It is better to check `street_address1 `, `street_address1 ` otherwise return empty string. 
Please change it.

Reply:
@YevhenShBolt there is no error here. Sometimes there is an address and sometimes not (prefetch shipping from the shopping cart page and probably geolocation, apple pay shipping and tax call..). Also the performance impact here is meaningless. I am leaving it as is.

Reply:
@YevhenShBolt Using `@` in this context costs about 0.005 ms. There is certainly a trade-off and a discussion to be had here, but I don't think it's about performance. A better argument against `@` is that it is against the `Magento2` PHPCS rules - we had to write a custom sniffer to get around that. I think for now, we should keep and use these.

Reply:
@matt-thomason @danac-gs 
1. I would probably agree that this suppressor is not so critical for this piece of code.
2. @matt-thomason  It is not clear for me why we should have our own rule if Magento provides recommendations on best practices and monitors the purity of the code. It also seems Magento checks the code with the same validator before loading it into its marketplace.

I think we should move `phpcs rule` discussion to another place. :-)